app/models/project/info_request_extension.rb

```diff
@@ -14,15 +14,20 @@ def classified
       where(awaiting_description: false)
     end
 
-    def extractable
-      where(described_state: EXTRACTABLE_STATES).
+    def extractable(project)
```

Comment:
Having to go `@project.info_requests.extractable(@project)` is a bit verbose.

Is there a reason we need the argument here? I assume it's added due to this module being used as an association extension.

I think we can use one of the `proxy_association`[^1] methods to get at the current `project` instance, so we'd end up with:

```ruby
# …
# TODO: figure out which proxy_association method is appropriate instead of placeholder "foo"
scope.where(project_submissions: { id: nil }).or(
  scope.where.not(project_submissions: { project: proxy_association.foo })
)
# …
```

I forget whether it's `#owner`, `#target`  or `#reflection` that's appropriate in this case.

[^1]: https://guides.rubyonrails.org/association_basics.html#association-extensions

Reply:
Yes, I came to the same conclusion when I saw the failing specs due to other calls to these method I hadn't updated. Hence the fixup using `@assoication.owner` but will look to replace with `proxy_assoication.onwer`.